src/ConsoleHost/Program.cs

```diff
@@ -18,20 +18,62 @@ static void Main(string[] args)
                 var fn = engine.EvaluateScriptText(@"(function() {
     echo('{0}, {1}!', 'Hello', 'world');
 })();");
-                fn.Invoke(Enumerable.Empty<JavaScriptValue>());
+                var rEcho = fn.Invoke(Enumerable.Empty<JavaScriptValue>());
+                Console.WriteLine("rEcho.Type: {0}", rEcho.Type); // Undefined
 
-                dynamic fnAsDynamic = fn;
-                fnAsDynamic.foo = 24;
-                dynamic global = engine.GlobalObject;
-                global.echo("{0}, {1}, via dynamic!", "Hey there", "world");
+                var createPerson = engine.EvaluateScriptText(@"(function(){ return { name: 'Cameron', age: 36 } })();");
+                var person = createPerson.Invoke(Enumerable.Empty<JavaScriptValue>());
+                Console.WriteLine("person.Type: {0}", person.Type); // Object
+                Console.WriteLine("person.GetType: {0}", person.GetType());
+                var prsn = (JavaScriptObject)person;
 
-                dynamic echo = global.echo;
-                echo("Whoa, {0}, that {1} {2}???", "world", "really", "worked");
-
-                foreach (dynamic name in global.Object.getOwnPropertyNames(global))
+                foreach (var mn in prsn.GetOwnPropertyNames())
                 {
-                    echo(name);
+                    Console.WriteLine("mn: {0}", mn);
                 }
+
+                var name = prsn.GetPropertyByName("name");
+                Console.WriteLine("name type: {0}", name.GetType());
+                Console.WriteLine("name: {0}", name.ToString());
+
+                var nm = prsn.GetPropertyByName("name");
+                Console.WriteLine("name type: {0}", nm.Type);
+                Console.WriteLine("name: {0}", nm.ToString());
+
+                // How to I get the Number value?
+                var age = prsn.GetPropertyByName("age");
+                Console.WriteLine("age type: {0}", age.Type);
+                //var a = age as JavaScriptNum
+                //Console.WriteLine("age: {0}", age);
+
+                // Is it possible to pass a JavaScriptObject back in?
+                var getName = engine.EvaluateScriptText(@"(function(person){ return person.name; });");
+                Console.WriteLine("getName type: {0}", getName.Type); // Function
+
+                var getName2 = getName.Invoke(Enumerable.Empty<JavaScriptValue>()); // JsErrorScriptException
+                Console.WriteLine("getName2 type: {0}", getName2.Type); // Function
```

Comment:
I got the function created by `EvaluateScriptText` then `Invoke`'d it to get the actual function. On my project, I just called `JsRunScript` [directly](https://github.com/ctaggart/TsAst/blob/0dd61989ae907988c56e7d6da2800820a76640f2/Program.fs#L107).